models/DataObject/ClassDefinition/Data/Objectbricks.php

```diff
@@ -620,7 +621,15 @@ public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
                             try {
                                 $key = $fd->getName();
                                 $getter = 'get' . ucfirst($key);
-                                $fd->checkValidity($item->$getter(), false, $params);
+
```

Comment:
hmm, I thing we should do it the same way as we do this check for class attributes, see [here](https://github.com/pimcore/pimcore/blob/b42dbbf37d10f6460662f5dc518b24d4028054b0/models/DataObject/Concrete.php#L133)

please adapt accordingly, because currently it is not checked, if inheritance is activated for the class or not, and if the field definition supports inheritance at all...